### PR TITLE
Add Endpoint list all Groups

### PR DESCRIPTION
**Pull request by pmanrubia**
## DO NOT MERGE BEFORE #96 
Only commit: 1ae92b3dffe1825d30d35753536e715523b5ecc8 applies

### Description

* Add endpoint to list all groups. 
* The list of groups can be filtered by `group_type`

### Samples

* List all groups

```terminal
curl "http://content-performance-manager.dev.gov.uk/groups.json?api_token=1234567" \
     -H "Content-Type: application/json"
```

* List all groups belonging to a type

```terminal
curl "http://content-performance-manager.dev.gov.uk/groups.json?api_token=1234567&group_type=inventory" \
     -H "Content-Type: application/json"
```

### CODE REVIEW

File: spec/controllers/groups_controller_spec.rb [hunk 1]
```diff
@@ -19,6 +19,23 @@
     end
   end
 
+  describe "GET #index" do
+    it "assigns all users as @users" do
```
Comment (chao-xian) [inline spec/controllers/groups_controller_spec.rb]: Again I think you mean `group`

Comment (pmanrubia) [inline spec/controllers/groups_controller_spec.rb]: Thank you. Updated in d3942a5

**Comment by pmanrubia** [comment]
@chao-xian I have addressed your comment. Please, let me know if you prefer to change any other thing.

File: spec/controllers/groups_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        group = create :group
+        get :show, params: { id: group.to_param }, format: :json
+
+        expect(response.status).to eq(401)
+      end
+    end
+  end
+
+  describe "GET #index" do
+    it "assigns all users as @groups" do
```
Comment (chao-xian) [inline spec/controllers/groups_controller_spec.rb]: It still says `users` in one place in the sentence.

Comment (pmanrubia) [inline spec/controllers/groups_controller_spec.rb]: Sorry, done now in 6da10c4

**Pull request merged**